lib/qesapdeployment.pm

```diff
@@ -1786,12 +1786,20 @@ sub qesap_is_job_finished {
 
     my $job_data = eval { decode_json($json_data) };
     if ($@) {
-        record_info("JSON error", "Failed to decode JSON data for job $job_id: $@");
-        return 0;    # Assume job is still running if we can't get its state
+        if ($json_data =~ /<h1>Page not found<\/h1>/) {
+            record_info(
+                "JOB NOT FOUND",
+                "Job $job_id was not found on the server " . get_required_var('OPENQA_HOSTNAME') .
+                  ". It may be deleted, from a different openqa server or from a manual deployment."
+            );
+        }
+        else {
+            record_info("OPENQA QUERY FAILED", "Failed to decode JSON data for job $job_id: $@");
+        }
+        return 0;    # assume job is still running if we can't get its info
     }
 
-    my $job_state = $job_data->{job}->{state} // 'running';    # assume job is running if unable to get status
-
+    my $job_state = eval { $job_data->{job}->{state} // 'running' } // 'running';    # assume job is running if unable to get status
```

Comment:
Why is the `eval` needed here?

Reply:
My idea was to avoid an exception in the case that, i.e., $job_data->{job} exists, but {job}->{state} does not. Not sure how `//` would handle this

Reply:
Ah, got it. I think it should be safe without the `eval`. Some tests:
```
acarvajal@airfox:~/git/qam-openqa-yml [master|✔] > perl
use strict;
use warnings;
my $r = { job => { state => 'running' } };
my $job_state = $r->{job}->{state} // 'else case';
print "job state: $job_state\n";
job state: running
###
acarvajal@airfox:~/git/qam-openqa-yml [master|✔] > perl
use strict;
use warnings;
my $r = { };
my $job_state = $r->{job}->{state} // 'else case';
print "job state: $job_state\n";
job state: else case
####
acarvajal@airfox:~/git/qam-openqa-yml [master|✔] > perl
use strict;
use warnings;
my $r = { job => { notstate => 'notrunning' } };
my $job_state = $r->{job}->{state} // 'else case';
print "job state: $job_state\n";
job state: else case
acarvajal@airfox:~/git/qam-openqa-yml [master|✔] > perl
use strict;
use warnings;
my $r = undef;
my $job_state = $r->{job}->{state} // 'else case';
print "job state: $job_state\n";
job state: else case
```
Seems like it behaves the same when any of `$job_data` or `$job_data->{job}` or `$job_data->{job}->{state}` is not defined.

Reply:
That feature is called autovivification and lets you check deeply nested structs without having to check each level in between for existance. Something I miss in other languages. But it can have surprising side effects as it creates the data structure in question :)
```perl
use Data::Dumper;
my %x;
if ($x{a}->{b}->{c}) { say "exists" }
say Dumper \%x
__END__
$VAR1 = {
          'a' => {
                   'b' => {}
                 }
        };
```

Reply:
That was really interesting! After evaluating this case, I think we can let perl safely create the structure if it doesn't exist - it's not gonna be used again, let alone for existence checks. Changed the code to remove `eval`